### DISCUSSION TITLE
Styles

### DISCUSSION THREAD

**Discussion by bwaltz** [ColbyCommunications/colby-college-theme#44] [General]
I figured I'd move a larger discussion about page content styles here instead of in individual issues.

Related issues:
https://github.com/ColbyCommunications/colby-college-theme/issues/41
https://github.com/ColbyCommunications/colby-college-theme/issues/40

We are finding that there aren't a breadth of styles for the numerous html elements that can exist within imported site content (h2 - h5, hrs, brs, lists of varying levels, links deep within lists, etc) and that the current styles are too specific to the top-level elements that were in the designs.

A few questions for you (and I by no means question your work style or development preferences, just curious).
-   I see the theme is generating two css files, tailwind.css and style.css, both seem to be used. I wonder if it would make it easier to just generate one with webpack. This way we can have everything in one place and won't have any style clashes.

In this case, we could do something like:

```
// style.scss
@tailwind base;
@tailwind components;
@tailwind utilities;
...other imported stylesheets

.inner-content-div {

  h3 {
    @apply text-lg py-4;
  }

  ul {
    @apply list-disc;

      >li>ul {
         @apply list-square;
     }
  }

// ...etc
}
```
then webpack would just use src/styles/style.scss as the entry point. Then we start to build some broader and consistent global styles.

- What does this code block do? https://github.com/ColbyCommunications/colby-college-theme/blob/master/src/twig/pages/page_with-sidebar.twig#L58

**Comment by bwaltz** [comment]
Are we using foundation? https://github.com/ColbyCommunications/colby-college-theme/blob/master/src/styles/global/_variables.scss#L1

**Comment by gregoryspragginsjr** [comment]
Hey @bwaltz 
No problem at all; I don't mind the questions. The two file styles are simply there to differentiate specific tailwind generated and miscellaneous styles. In a typical work environment, styles would be the main stylesheet. Some of the stuff in there is boilerplate / old and simply needs to be cleaned out (like foundation as you've mentioned). What you're seeing on `page_with-sidebar.twig` are the tailwind classes that dictate those behaviors a long with spacings rules. They are specific like that to to ensure that loose wysiwyg styles do not conflict with styles being generated by specific components that may also live within a `page_with-sidebar` content area, and they are formated like that because we're simply trying to stay within tailwind's preferred styling applications via utility classes. The goal is to never have to resort to plain sass unless it's an edge case like the form embeds from gravity forms or the contact page) Otherwise, it kinda defeats the purpose of the tailwind framework in my opinion.

For those missing element styles, we can easily add them (h1-h3 and all list styles should be present already. I can add h4-h6 based on what styles that are still available from our styleguide and we can whip up an hr). For the instances in which styles arent getting hit due to being nested a little too deep, we can loosen specificity or I can apply some more rules. Ill have an update soon.

**Comment by bwaltz** [comment]
Thanks for the clarity, @gregoryspragginsjr. Yeah tailwind can be finicky because it really doesn't give you any default styles like bootstrap and others, so you end up needing to do that yourself. It's tough, especially with gutenberg and imported content, because you don't know what content situations you'll run into and you may need to write a lot of miscellaneous rules, unless you emulate some bootstrap-like global styles, I've found.

Yeah it does defeat the purpose of tailwind to some degree, but WP and gutenberg allows for so much flexibility! It's tough to keep things under control.

**Comment by bwaltz** [comment]
@gregoryspragginsjr just curious. Are we disabling WP native block styles? I don't see the blocks.css file being loaded. When I switch to the twentyseventeen theme, for example, I see:

<img width="1396" alt="Screen Shot 2022-11-14 at 3 23 42 PM" src="https://user-images.githubusercontent.com/2333967/201758658-6581f5b1-cc73-4dee-86d7-be43aae0ddda.png">

**Comment by gregoryspragginsjr** [comment]
@bwaltz We aren't manually disabling it but `blocks.css` I do not believe is the Gutenberg native stylesheet. It is the specific block styling for the Twenty Seventeen theme.
